### DISCUSSION TITLE
Unexpected /v2/ being added to path when using JFrog Artifactory SaaS as Docker Registry

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#17201] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.159.5

### Please select which platform you are using if self-hosting.

Bitbucket Cloud (bitbucket.org)

### If you're self-hosting Renovate, tell us what version of the platform you run.

Bitbucket Cloud

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

We've been trying to configure JFrog Artifactory SaaS as one of our Docker `registryUrls`

Example configuration fragment
```
packageRules: [
    {
      matchDatasources: ["docker"],
      registryUrls: ["https://companyname.jfrog.io/artifactory/api/docker/virtual-docker/v2/"]
    }
  ],
```

When reviewing the debug logs, I can see the following requests are being made
```
DEBUG: http statistics (repository=wfscorp/renovate-demo-repo-onboarded)
"urls": {
  ...,
  "https://companyname.jfrog.io/v2/artifactory/api/docker/virtual-docker/v2/library/node/tags/list (GET,401)": 1,
  "https://companyname.jfrog.io/v2/artifactory/api/docker/virtual-docker/v2/library/node/tags/list (GET,404)": 1,
  "https://companyname.jfrog.io/v2/artifactory/api/docker/virtual-docker/v2/node/tags/list (GET,401)": 1,
  "https://companyname.jfrog.io/v2/artifactory/api/docker/virtual-docker/v2/node/tags/list (GET,404)": 1,
  ...
 }
 ```
 
Note the URL has an additional `/v2/` path inserted after the host name and is therefore resulting in a HTTP 401 or HTTP 404.

When testing via Postman/Curl, I can confirm that data is received when accessing the following endpoint

- https://companyname.jfrog.io/artifactory/api/docker/virtual-docker/v2/library/node/tags/list

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
this is expected, as the docker API requires it to com first in path. docker and kubernetes behave same.

so move `/v2/` to beginning in your registry url.

**Reply by setchy** [reply]
Ah yes, thank you @viceice - I was incorrectly trying to reference a specific virtual jfrog repository  

Sharing some more detail for others in future...
 
Setting `registryUrls: ["https://companyname.jfrog.io"]` worked, but only for  Dockerfile `FROM` statements which were fully qualified (eg: `FROM companyname.jfrog.io/virtual-docker/library/node:17-alpine`).  Any shorthand `FROM` statements (eg: `FROM node:17-alpine`) we unsuccessful resolved (JFrog returns HTTP 404).  

Setting multiple `registryUrls` is also not supported for docker datasources.  Example error/warning below
```
Excess registryUrls found for datasource lookup - using first configured only
```

🎉 What did work was:

- leaving the docker registryUrls unchanged (ie: not setting any custom registries)
- using either shorthand or fully qualified `FROM` statements
- configuring a hostRules for `companyname.jfrog.io`.  

This allowed shorthand images to be resolved via docker.io, while fully qualified companyname.jfrog.io were resolved via JFrog Artifactory using the authentication provided in hostRules.
